content-src/components/Sections/Sections.jsx

```diff
@@ -137,14 +137,21 @@ export class Section extends React.PureComponent {
     const maxCards = maxCardsPerRow * numRows;
     const maxCardsOnNarrow = CARDS_PER_ROW_DEFAULT * numRows;
 
+    const {pocketCta, isUserLoggedIn} = Pocket || {};
+    const {useCta} = pocketCta || {};
+
+    // Don't display anything until we have a definitve result from Pocket,
+    // to avoid a flash of logged out state while we render.
+    const isPocketLoggedInDefined = (isUserLoggedIn === true || isUserLoggedIn === false);
+
     const shouldShowPocketCta = (id === "topstories" &&
-      Pocket.pocketCta.useCta && !Pocket.isUserLoggedIn);
+      useCta && isUserLoggedIn === false);
 
-    // Show topics only for top stories and if it's not initialized yet (so
-    // content doesn't shift when it is loaded) or has loaded with topics
+    // Show topics only for top stories and if it has loaded with topics.
+    // The classs .top-stories-bottom-container ensures content doesn't shift as things load.
     const shouldShowTopics = (id === "topstories" &&
-      (!topics || topics.length > 0) &&
-      !shouldShowPocketCta);
+      (topics && topics.length > 0) &&
+      ((useCta && isUserLoggedIn === true) || (!useCta && isPocketLoggedInDefined)));
```

Comment:
I'm essentially trying to sort out this logic so we don't display anything until we know enough of what to display.

Like I mentioned above, the content has a placecholder, and the "more recs" link is still displayed. I think this should be fast as it's only waiting for the pocket jsm to return if the user is logged in or not.